apps/financial-aid/web-veita/src/routes/Search/search.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useMemo, useState } from 'react'
+
+import {
+  ApplicationOverviewSkeleton,
+  LoadingContainer,
+  TableHeaders,
+  SearchSkeleton,
+  TableBody,
+  TextTableItem,
+  Name,
+  State,
+} from '@island.is/financial-aid-web/veita/src/components'
+import { Text, Box } from '@island.is/island-ui/core'
+
+import * as tableStyles from '../../sharedStyles/Table.css'
+import * as styles from './search.css'
+import cn from 'classnames'
+import { SearchApplicationQuery } from '@island.is/financial-aid-web/veita/graphql'
+import { useLazyQuery } from '@apollo/client'
+import {
+  getMonth,
+  Routes,
+  Application,
+} from '@island.is/financial-aid/shared/lib'
+import router from 'next/router'
+import { useRouter } from 'next/router'
+
+export const Search = () => {
+  const router = useRouter()
+
+  const [searchNationalId, setSearchNationalId] = useState<string>(
+    router?.query?.search as string,
+  )
+
+  useEffect(() => {
+    if (sanitize(searchNationalId).length === 10) {
+      getApplications({
+        variables: {
+          input: { nationalId: sanitize(searchNationalId) },
+        },
+      })
+    }
+  }, [])
+
+  const sanitize = (n: string) => n.replace(/[^0-9]/g, '')
+
+  const isValid = (nationalId: string): boolean => {
```

Comment:
This could also be in utils looks like a very general function, which could be nice to use in other places